setup.py

```diff
@@ -36,8 +37,20 @@
 with open("README.md") as file:
     long_description = file.read()
 
+OMP_CONFIG = defaultdict(
+    lambda: ["-fopenmp"], {"unix": ["-fopenmp"], "msvc": ["/openmp"]}
+)
+CPP14_CONFIG = defaultdict(
+    lambda: ["-std=c++14"], {"unix": ["-std=c++14"], "msvc": ["/std:c++14"]}
+)
+CPP03_CONFIG = defaultdict(
+    lambda: ["-std=c++03"], {"unix": ["-std=c++03"], "msvc": ["/std:c++03"]}
+)
+
+_COMPILER = get_default_compiler()
+
 if check_for_openmp():
```

Comment:
`check_for_openmp()` makes some assumptions about openmp flags. As a result your code will only be called only if `-fopenmp` works, otherwise it will default to `[]`

Reply:
yes, the assumption about flags (tested and applied) should be in one place, e.g. `check_for_openmp`. we currently duplicate the flags (and assumptions).